### PR TITLE
pymetno: init at 0.5.0

### PR DESCRIPTION
**Pull request by flyfloh**
Another library used by Home-Assistant. This one is for talking to the met.no service and getting the weather.

It seems the tests for `aiohttp` break when building this for python 3.8, but it works for 3.6 and 3.7.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pymetno/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "PyMetno";
+  version = "0.5.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "e532544495200210407e3d68f719c271435da6f3bfe696e708b1d5d603a21948";
+  };
```
Comment (jonringer) [inline pkgs/development/python-modules/pymetno/default.nix]:
```
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
/nix/store/lkll7y216pk69jl93kizxgmi9hnpv5wy-python3.7-PyMetno-0.5.0
```

no tests packages in pypi sdist, please fetch from github.

Tests are critical to ensuring the package isn't broken.

Comment (flyfloh) [reply pkgs/development/python-modules/pymetno/default.nix]: Done. Unfortunately there are not tests on Github either.

Comment (jonringer) [reply pkgs/development/python-modules/pymetno/default.nix]: :(

**Comment by jonringer** [comment]
unfortunatley, I don't think this is a great candidate for nixpkgs (personal project, no tests), however, https://github.com/nix-community/NUR  would be a good candidate if you wanted to share with others.

**Comment by Mic92** [comment]
Are you sure? It's used within home-assistant, which we have packaged.

**Comment by flyfloh** [comment]
I'd be happy to submit a PR to NUR if that's the place to put it. However as @Mic92 pointed out home-assistant depends on this library to get the weather from the met.no service. Wouldn't HA be much harder to use when parts of it depend on NUR?

**Comment by jonringer** [comment]
I don't currently see it in nixpkgs.

I would be okay with doing `pythonImportsCheck` on the majority of importable modules. Just want to ensure that it's actually usable when doing dependency bumps. Unit tests are much more preferable as they give some notion of correctness.

**Comment by flyfloh** [comment]
> I would be okay with doing `pythonImportsCheck` on the majority of importable modules.

I don't quite get that. What does `pythonImportsCheck` do? `grep`ing for it didn't really enlighten me. Seems like most packages that use it put themselves in there.

Should I do something like this?

```
pythonImportsCheck = [ "pymetno"];
```

**Comment by jonringer** [comment]
```
pythonImportsCheckPhase () {
    echo "Executing pythonImportsCheckPhase"

    if [ -n "$pythonImportsCheck" ]; then
        echo "Check whether the following modules can be imported: $pythonImportsCheck"
        ( cd $out && eval "@pythonCheckInterpreter@ -c 'import os; import importlib; list(map(lambda mod: importlib.import_module(mod), os.environ[\"pythonImportsCheck\"].split()))'" )
    fi
}
```
It will essentially do `python -c 'import XXXX'` to each of the modules you list. This will at least cause the python interpreter to import/evaluate all the modules that it needs at the beginning of the files.

**Comment by Mic92** [comment]
Let's go with this solution.

**Comment by flyfloh** [comment]
I have updated this to use `pythonImportsCheck`. Thanks for the explanation @jonringer.

File: pkgs/development/python-modules/pymetno/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "00v2r3nn48svni9rbmbf0a4ylgfcf93gk2wg7qnm1fv1qrkgscvg";
+  };
+
+  propagatedBuildInputs = [ aiohttp async-timeout pytz xmltodict ];
+
+  pythonImportsCheck = [ "metno"];
+
+  meta = with stdenv.lib; {
+    description = "A library to communicate with the met.no api";
+    homepage = https://github.com/Danielhiversen/pyMetno/;
```
Comment (jonringer) [inline pkgs/development/python-modules/pymetno/default.nix]:
```suggestion
    homepage = "https://github.com/Danielhiversen/pyMetno/";
```

**Review by jonringer — COMMENTED** [review]
rfc 45, otherwise LGTM'

```
[3 built, 66 copied (278.0 MiB), 60.6 MiB DL]
https://github.com/NixOS/nixpkgs/pull/78132
1 package built:
python37Packages.pymetno
```

**Pull request merged**